pkg/graveler/graveler.go

```diff
@@ -1684,53 +1718,52 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 		if params.SourceMetaRange != nil {
 			commit.MetaRangeID = *params.SourceMetaRange
 		} else {
-			changes, err := g.StagingManager.List(ctx, branch.StagingToken, ListingMaxBatchSize)
+			changes, err := g.listSealedTokens(ctx, branch)
 			if err != nil {
-				return "", fmt.Errorf("staging list: %w", err)
+				return nil, err
 			}
 			defer changes.Close()
 
 			commit.MetaRangeID, _, err = g.CommittedManager.Commit(ctx, storageNamespace, branchMetaRangeID, changes)
 			if err != nil {
-				return "", fmt.Errorf("commit: %w", err)
+				return nil, fmt.Errorf("commit: %w", err)
 			}
 		}
 
 		// add commit
 		newCommit, err := g.RefManager.AddCommit(ctx, repositoryID, commit)
 		if err != nil {
-			return "", fmt.Errorf("add commit: %w", err)
+			return nil, fmt.Errorf("add commit: %w", err)
 		}
-		err = g.RefManager.SetBranch(ctx, repositoryID, branchID, Branch{
-			CommitID:     newCommit,
-			StagingToken: generateStagingToken(repositoryID, branchID),
-		})
-		if err != nil {
-			return "", fmt.Errorf("set branch commit %s: %w", newCommit, err)
+
+		for _, st := range branch.SealedTokens {
+			err = g.StagingManager.Drop(ctx, st)
+			if err != nil {
+				g.log.WithContext(ctx).WithFields(logging.Fields{
+					"repository_id": repositoryID,
+					"branch_id":     branchID,
+					"commit_id":     branch.CommitID,
+					"message":       params.Message,
+					"staging_token": branch.StagingToken,
+				}).Error("Failed to drop staging data")
+			}
 		}
-		err = g.StagingManager.Drop(ctx, branch.StagingToken)
-		if err != nil {
-			g.log.WithContext(ctx).WithFields(logging.Fields{
-				"repository_id": repositoryID,
-				"branch_id":     branchID,
-				"commit_id":     branch.CommitID,
-				"message":       params.Message,
-				"staging_token": branch.StagingToken,
-			}).Error("Failed to drop staging data")
-		}
-		return newCommit, nil
+
+		newCommitID = newCommit
+		branch.CommitID = newCommit
```

Comment:
I think you need to remove the sealedTokens here

Reply:
done